### PR TITLE
docs: Add Vite Setup guide

### PR DESCRIPTION
**Pull request by thekip**
# Description

Docs for Vite integration

## Types of changes

[//]: # (What types of changes does your code introduce to Lingui?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by thekip** [comment]
@andrii-bodnar i have no idea why docs suite fail. May be restart of the WF would help?

**Comment by andrii-bodnar** [comment]
@thekip restarted. It's some weird random issue

File: website/docs/tutorials/setup-vite.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+          babel: {
+            plugins: ["macros"]
+          }
+        }),
+        lingui()
+      ]
+    });
+   ```
+
+## Setup with [@vitejs/plugin-react-swc](https://www.npmjs.com/package/@vitejs/plugin-react-swc)
```
Comment (andrii-bodnar) [inline website/docs/tutorials/setup-vite.md:40]:
```suggestion
## Setup with [@vitejs/plugin-react-swc](https://www.npmjs.com/package/@vitejs/plugin-react-swc) {#setup-with-vitejs-plugin-react-swc}
```

File: website/docs/tutorials/setup-vite.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+    export default defineConfig({
+      plugins: [
+        react({
+          plugins: [["@lingui/swc-plugin", {}]]
+        }),
+        lingui(),
+      ]
+    });
+   ```
```
Comment (andrii-bodnar) [inline website/docs/tutorials/setup-vite.md:65]:
```suggestion

   ```ts title="vite.config.ts"
   import { defineConfig } from "vite";
   import react from "@vitejs/plugin-react-swc";
   import lingui from "@lingui/vite-plugin";

   export default defineConfig({
     plugins: [
       react({
         plugins: [["@lingui/swc-plugin", {}]]
       }),
       lingui(),
     ]
   });
   ```

```

File: website/docs/tutorials/setup-vite.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+## Further Setup
+1. Create `lingui.config.js` file with LinguiJS configuration in root of your project (next to `package.json`). Replace `src` with a directory name where you have source files:
```
Comment (andrii-bodnar) [inline website/docs/tutorials/setup-vite.md:67]:
```suggestion

1. Create a `lingui.config.js` file with LinguiJS configuration in the root of your project (next to `package.json`). Replace `src` with a directory name where you have source files:

```

File: website/docs/tutorials/setup-vite.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+       }],
+       format: "po"
+    }
+  ```
+
+  PO format is recommended for message catalogs, and could be compiled on the fly thanks to `@lingui/vite-plugin`.
+
+  See [`format`](/docs/ref/catalog-formats.md) documentation for other available formats.
+
+2. Add following scripts to your `package.json`:
```
Comment (andrii-bodnar) [inline website/docs/tutorials/setup-vite.md:84]:
```suggestion
2. Add the following scripts to your `package.json`:
```

File: website/docs/tutorials/setup-vite.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    │ cs       │     0       │   0     │
+    │ en       │     0       │   0     │
+    │ fr       │     0       │   0     │
+    └──────────┴─────────────┴─────────┘
+
+    (use "lingui extract" to update catalogs with new messages)
+    (use "lingui compile" to compile catalogs for production)
+    ```
+
+   That command should create `.po` catalogs in `src/locales/` folder.
```
Comment (andrii-bodnar) [inline website/docs/tutorials/setup-vite.md:118]:
```suggestion
   This command should create `.po` catalogs in the `src/locales/` folder:
```

File: website/docs/tutorials/setup-vite.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+       i18n.load(locale, messages)
+       i18n.activate(locale)
+    }
+   ```
+
+See the [guide about dynamic loading catalogs](/docs/guides/dynamic-loading-catalogs.md) for more info.
+
+See [Vite's official documentation](https://vitejs.dev/guide/features.html#dynamic-import) for more info about Vite dynamic imports.
+
+Congratulations! You've successfully set up project with LinguiJS. Now it's good time to follow [React tutorial](/docs/tutorials/react.md) or read about [ICU Message Format](/docs/ref/message-format.md) which is used in messages.
```
Comment (andrii-bodnar) [inline website/docs/tutorials/setup-vite.md:143]:
```suggestion
Congratulations! You've successfully set up a Vite project with LinguiJS. Now it's a good time to follow [React tutorial](/docs/tutorials/react.md) or read about [ICU Message Format](/docs/ref/message-format.md) which is used in messages.
```

File: website/docs/tutorials/setup-vite.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+The integration for LinguiJs for Vite:
+- Supports both [@vitejs/plugin-react](https://www.npmjs.com/package/@vitejs/plugin-react) and [@vitejs/plugin-react-swc](https://www.npmjs.com/package/@vitejs/plugin-react-swc)
+- Compiling `.po` catalogs on the fly
+
+## Setup with [@vitejs/plugin-react](https://www.npmjs.com/package/@vitejs/plugin-react)
+
+`@vitejs/plugin-react` uses Babel to transform your code. LinguiJS rely on `babel-plugin-macros` to compile JSX to [ICU Message Format](/docs/ref/message-format.md) and for automatic ID generation.
+
+1.  Install `@lingui/cli`, `babel-plugin-macros` as development dependencies and `@lingui/macro`, `@lingui/react` as a runtime dependency.
```
Comment (andrii-bodnar) [inline website/docs/tutorials/setup-vite.md:11]:
```suggestion
1.  Install `@lingui/cli`, `babel-plugin-macros` as development dependencies and `@lingui/macro`, `@lingui/react` as a runtime dependency:
```

File: website/docs/tutorials/setup-vite.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+        lingui()
+      ]
+    });
+   ```
+
+## Setup with [@vitejs/plugin-react-swc](https://www.npmjs.com/package/@vitejs/plugin-react-swc)
+
+`@vitejs/plugin-react-swc` uses [SWC](https://swc.rs/) to transform your code, which is 20x faster than Babel. LinguiJS rely on [@lingui/swc-plugin](/ref/swc-plugin) to compile JSX to [ICU Message Format](/docs/ref/message-format.md) and for automatic ID generation.
+
+1.  Install `@lingui/cli`, `@lingui/swc-plugin` as development dependencies and `@lingui/macro`, `@lingui/react` as a runtime dependency.
```
Comment (andrii-bodnar) [inline website/docs/tutorials/setup-vite.md:44]:
```suggestion
1.  Install `@lingui/cli`, `@lingui/swc-plugin` as development dependencies and `@lingui/macro`, `@lingui/react` as a runtime dependency:
```

File: website/docs/tutorials/setup-vite.md [hunk 9]
```diff
@@ -0,0 +1,146 @@
+# Setup with [Vite](https://vitejs.dev/guide/)
```
Comment (thekip) [inline website/docs/tutorials/setup-vite.md:1]:
```suggestion
# Setup with Vite
```

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**